crates/compiler/test_syntax/src/minimize.rs

```diff
@@ -0,0 +1,204 @@
+use crate::test_helpers::{Input, InputKind};
+use bumpalo::Bump;
+use roc_parse::{ast::Malformed, remove_spaces::RemoveSpaces};
+
+pub fn print_minimizations(text: &str, kind: InputKind) {
+    let Some(original_error) = round_trip_once_and_extract_error(text, kind) else {
+        eprintln!("No error found");
+        return;
+    };
+
+    eprintln!("Error found: {}", original_error);
+    eprintln!("Proceeding with minimization");
+
+    let mut s = text.to_string();
+
+    loop {
+        let mut found = false;
+        for update in candidate_minimizations(s.clone()) {
+            let mut new_s = String::with_capacity(s.len());
+            let mut offset = 0;
+            for (start, end, replacement) in update.replacements.clone() {
+                new_s.push_str(&s[offset..start]);
+                new_s.push_str(&replacement);
+                offset = end;
+            }
+            new_s.push_str(&s[offset..]);
+
+            assert!(
+                new_s.len() < s.len(),
+                "replacements: {:?}",
+                update.replacements
+            );
+
+            if let Some(result) = round_trip_once_and_extract_error(&new_s, kind) {
+                if result == original_error {
+                    eprintln!("Successfully minimized, new length: {}", new_s.len());
+                    s = new_s;
+                    found = true;
+                    break;
+                }
+            }
+        }
+
+        if !found {
+            eprintln!("No more minimizations found");
+            break;
+        }
+    }
+
+    eprintln!("Final result:");
+    println!("{}", s);
+}
+
+fn round_trip_once_and_extract_error(text: &str, kind: InputKind) -> Option<String> {
+    let input = kind.with_text(text);
+    let res = std::panic::catch_unwind(|| round_trip_once(input));
+
+    match res {
+        Ok(res) => res,
+        Err(e) => {
+            if let Some(s) = e.downcast_ref::<&'static str>() {
+                return Some(s.to_string());
+            }
+            if let Some(s) = e.downcast_ref::<String>() {
+                return Some(s.clone());
+            }
+            Some("Panic during parsing".to_string())
+        }
+    }
+}
+
+fn round_trip_once(input: Input<'_>) -> Option<String> {
+    let arena = Bump::new();
+
+    let actual = match input.parse_in(&arena) {
+        Ok(a) => a,
+        Err(e) => {
+            return Some(format!(
+                "Initial parse failed: {:?}",
+                e.remove_spaces(&arena)
+            ))
+        } // todo: strip pos info, use the error
```

Comment:
Another TODO

Reply:
So. Some history. `minimize` is actually a thing that I've had locally for multiple years, for working on various diffs.

It turns out that in that intervening time, this todo has actually been fixed!

(that's what `e.remove_spaces(&arena)` above is doing)